novawallet/Common/Protocols/Ramp/RampPresentable.swift

```diff
@@ -0,0 +1,116 @@
+import Foundation
+
+protocol RampPresentable {
+    func showRampAction(
+        from view: ControllerBackedProtocol?,
+        action: RampAction,
+        delegate: RampDelegate
+    )
+
+    func showOnRampProviders(
+        from view: ControllerBackedProtocol?,
+        actions: [RampAction],
+        assetSymbol: AssetModel.Symbol,
+        delegate: RampDelegate
+    )
+
+    func showOffRampProviders(
+        from view: ControllerBackedProtocol?,
+        actions: [RampAction],
+        assetSymbol: AssetModel.Symbol,
+        delegate: RampDelegate
+    )
+
+    func presentOnRampDidComplete(
+        view: ControllerBackedProtocol?,
+        locale: Locale
+    )
+
+    func presentOffRampDidComplete(
+        view: ControllerBackedProtocol?,
+        locale: Locale
+    )
+}
+
+extension RampPresentable {
+    func showRampAction(
+        from view: ControllerBackedProtocol?,
+        action: RampAction,
+        delegate: RampDelegate
+    ) {
+        guard let purchaseView = RampViewFactory.createView(
+            for: action,
+            delegate: delegate
+        ) else {
+            return
+        }
+        purchaseView.controller.modalPresentationStyle = .fullScreen
+        view?.controller.present(purchaseView.controller, animated: true)
+    }
+
+    func showOnRampProviders(
+        from view: ControllerBackedProtocol?,
+        actions: [RampAction],
+        assetSymbol: AssetModel.Symbol,
+        delegate: RampDelegate
+    ) {
+        guard let onRampProvidersView = SelectRampProviderViewFactory.createView(
+            providerType: .onramp,
```

Comment:
Intuitively it feels like we can derive provider type from ramp actions. It looks strange that we have RampAction but still separating flows between OnRamp and OffRamp